--- a/glibmus-src/fenv.cpp
+++ b/glibmus-src/fenv.cpp
@@ -31,1 +31,5 @@
-    *flagp = env.__status_word & excepts;
+    #if defined(__aarch64__)
+        *flagp = env.__fpsr & excepts;
+    #else
+        *flagp = env.__status_word & excepts;
+    #endif

--- a/glibmus-src/fenv.cpp
+++ b/glibmus-src/fenv.cpp
@@ -51,1 +51,5 @@
-    env.__status_word = (env.__status_word & ~excepts) | (*flagp & excepts);
+    #if defined(__aarch64__)
+        env.__fpsr = (env.__fpsr & ~excepts) | (*flagp & excepts);
+    #else
+        env.__status_word = (env.__status_word & ~excepts) | (*flagp & excepts);
+    #endif

--- a/glibmus-src/fenv.cpp
+++ b/glibmus-src/fenv.cpp
@@ -76,1 +76,5 @@
-    env.__control_word &= ~(excepts & FE_ALL_EXCEPT);
+    #if defined(__aarch64__)
+        env.__fpcr &= ~(excepts & FE_ALL_EXCEPT);
+    #else
+        env.__control_word &= ~(excepts & FE_ALL_EXCEPT);
+    #endif

--- a/glibmus-src/fenv.cpp
+++ b/glibmus-src/fenv.cpp
@@ -101,1 +101,5 @@
-    env.__control_word |= (excepts & FE_ALL_EXCEPT);
+    #if defined(__aarch64__)
+        env.__fpcr |= (excepts & FE_ALL_EXCEPT);
+    #else
+        env.__control_word |= (excepts & FE_ALL_EXCEPT);
+    #endif
